bentoml/cli/click_utils.py

```diff
@@ -56,6 +64,49 @@ def _echo(message, color="reset"):
     click.secho(message, fg=color)
 
 
+def _generate_track_event_name(command, sub_cmd_group=None):
+    if sub_cmd_group is None:
+        if command == 'open_api_spec':
+            command = 'open-api-spec'
+        return command
+
+    # Format deployment related event names to match with existing event names
+    if sub_cmd_group in DEPLOYMENT_CMD_GROUPS:
+        if command == 'list_deployments':
+            command = 'list'
+        event_name = f'deploy-{command}'
+    else:
+        # Special case for bentoml config set. The name of func is `set_command`. It
+        # will be changed to 'set'
+        if command == 'set_command':
+            command = 'set'
+        event_name = f'{sub_cmd_group}-{command}'
+
+    return event_name
+
+
+def _get_deployment_platform(command_group):
+    if command_group in DEPLOYMENT_CMD_GROUPS:
+        return command_group.upper()
+    else:
+        return None
+
+
+def _get_sub_cmd_group(modules):
+    module_list = modules.split('.')
```

Comment:
this is not guaranteed, we can easily break this after some internal code refactoring

Also the event name being recorded is not tested in the unit test - as commented above, you need to verify that `track` gets called with expected parameters